app/api/items/navigation_data_mapper.go

```diff
@@ -54,15 +54,27 @@ func getRawNavigationData(dataStore *database.DataStore, rootID, groupID, attemp
 				NULL AS score_computed, NULL AS validated, NULL AS started_at, NULL AS latest_activity_at,
 				NULL AS allows_submissions_until, NULL AS ended_at`).
 		Joins(`
-			JOIN results ON results.participant_id = ? AND results.attempt_id = ? AND
-				results.item_id = items.id AND results.started`, groupID, attemptID)
+			JOIN results
+				ON results.participant_id = ?
+			 AND results.attempt_id = ?
+			 AND results.item_id = items.id
+			 AND results.started
+		`, groupID, attemptID)
 	service.MustNotBeError(itemsQuery.Error())
 
 	hasVisibleChildrenQuery := dataStore.Permissions().MatchingGroupAncestors(groupID).
 		WherePermissionIsAtLeast("view", "info").
 		Joins("JOIN items_items ON items_items.child_item_id = permissions.item_id").
+		Joins(`
+			JOIN items AS child_items
+			  ON child_items.id = items_items.child_item_id
+			 AND (items.type = "Skill" AND child_items.type = "Skill")
+				OR items.type <> "Skill"
```

Comment:
Don't you mean 
```suggestion
			  ON child_items.id = items_items.child_item_id
			 AND ((items.type = "Skill" AND child_items.type = "Skill")
				OR items.type <> "Skill")
```
?
If so, couldn't you add a test to detect a similar regression in the future?

Reply:
fixed

Wow, this one was tricky.

I couldn't come with a test to cover the case because without the extra parenthesis, it still works. But the fact it works without it is more a coincidence.

It is interpreted as:
```
JOIN items AS child_items
  ON (child_items.id = items_items.child_item_id AND (items.type = "Skill" AND child_items.type = "Skill"))
  OR items.type <> "Skill"
```

When `items.type="Skill"`, it is as expected. But when `items.type<>"Skill"`, it doesn't make any restriction on the other joined tables, so it has no effect. Which is also what we want because a Chapter cannot have Skills as children.

But the issue is that without restricting the other joined tables, it internally makes a cross product between the tables, which can be very slow and consume a lot of memory.

So, I added the parenthesis, but couldn't make a test for verification because it doesn't change the behavior.